### PR TITLE
10173 Misc date cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#10173

*Description of changes:*
- renamed createUtcDateFromStr -> yyyyMmDdToDate 
- renamed dateStrToPacificDate -> yyyyMmDdToPacificDate
- deleted unused date mixin variable and method
- added date-mixin test file
- added legal-api-mixin test file
- added mixin-tester
- misc cleanup
- app version: 3.0.30

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/mixins/name-request-mixin.ts [hunk 1]
```diff
@@ -97,14 +96,15 @@ export default class NameRequestMixin extends Mixins(DateMixin) {
...
-    if (nr.names?.length > 0) {
-      return nr.names.find(name => [NameRequestStates.APPROVED, NameRequestStates.CONDITION].includes(name.state)).name
+  getNrApprovedName (nr: any): string {
+    if (nr?.names?.length > 0) {
+      return nr.names
+        .find(name => [NameRequestStates.APPROVED, NameRequestStates.CONDITION].includes(name.state))?.name
     }
-    return '' // should never happen
+    return null // should never happen
   }
```
Comment (severinbeauvais) [inline src/mixins/name-request-mixin.ts:108]: I changed this method to be consistent across all Entity UIs.

File: tests/unit/date-mixin.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import Vue from 'vue'
```
Comment (severinbeauvais) [inline tests/unit/date-mixin.spec.ts:1]: Copied here from Filings UI, and deleted the methods not in this project.

File: tests/unit/legal-api-mixin.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import Vue from 'vue'
```
Comment (severinbeauvais) [inline tests/unit/legal-api-mixin.spec.ts:1]: Copied here from Filings UI, and updated to the methods in this project.

**Comment by severinbeauvais** [comment]
Ready for review. This is mostly clean / sync methods with other UIs. Should be safe to merge/deploy.

File: src/mixins/date-mixin.ts [hunk 4]
```diff
@@ -58,17 +56,20 @@ export default class DateMixin extends Mixins(CommonMixin) {
   }
 
   /**
-   * Creates and returns a new Date object in UTC, given parameters in Pacific timezone in string format YYYY-mm-dd.
+   * Converts a date string (YYYY-MM-DD) to a Date object at 12:00:00 am Pacific time.
+   * @example 2021-11-22 -> 2021-11-22T08:00:00.00Z
    */
-  createUtcDateFromStr (dateStr: string): Date {
-    // safety check
+  yyyyMmDdToDate (dateStr: string): Date {
```
Comment (argush3) [inline src/mixins/date-mixin.ts:62]: thanks for cleaning this function up!

Comment (severinbeauvais) [reply src/mixins/date-mixin.ts:62]: It turns out you and I both implemented this (in different projects). I took the best of both code :)

File: tests/unit/date-mixin.spec.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    expect(vm.dateToYyyyMmDd(new Date('2021-01-01 00:00:00 PST'))).toBe('2021-01-01')
+    expect(vm.dateToYyyyMmDd(new Date('2021-01-01 23:59:59 PST'))).toBe('2021-01-01')
+    expect(vm.dateToYyyyMmDd(new Date('2021-07-01 00:00:00 PDT'))).toBe('2021-07-01')
+    expect(vm.dateToYyyyMmDd(new Date('2021-07-01 23:59:59 PDT'))).toBe('2021-07-01')
+    // verify that Eastern is correctly converted to Pacific
+    expect(vm.dateToYyyyMmDd(new Date('2021-01-01 02:00:00 EST'))).toBe('2020-12-31')
+    expect(vm.dateToYyyyMmDd(new Date('2021-01-01 03:00:00 EST'))).toBe('2021-01-01')
+    expect(vm.dateToYyyyMmDd(new Date('2021-07-01 02:00:00 EDT'))).toBe('2021-06-30')
+    expect(vm.dateToYyyyMmDd(new Date('2021-07-01 03:00:00 EDT'))).toBe('2021-07-01')
+  })
```
Comment (argush3) [inline tests/unit/date-mixin.spec.ts:42]: something very soothing about look at this test 😋

**Review by argush3 — APPROVED** [review]
LGTM!

**Review by cameron-freshworks — APPROVED** [review]

**Pull request merged**